### ISSUE TITLE
Modifier Remover does not work in a 1.20 smithing table

### ISSUE THREAD

**Issue by Philbertz** [Robocraft999/CreateGoggles#26]
In 1.20.1 the smithing table has a third slot for the Smithing template, and therefore I can't remove the goggles from my diamond helmet with a modifier remover. It also means that I can't upgrade the helmet to netherite at all.

**Comment by Robocraft999** [comment]
You dont need a armor template to remove the goggles

**Comment by justinw2347** [comment]
with or without an armor template i am still unable to remove the goggles with the modifier remover

**Comment by SgtKaneki** [comment]
still present

**Comment by Robocraft999** [comment]
> still present

more info please. What minecraft version, fabric or forge and do you have the config option "enableExperimentalFeatures" enabled and what item do you try to apply it on?

**Comment by SgtKaneki** [comment]
Fabric version: 0.15.11
Minecraft version: 1.20.1
Mod version: 0.5.5g

enableExperimentalFeatures = true

The goggle was combined with a netherite helmet (worked), but Removing in Smithing Table with a Modifier Remover doesn't worked. You can put both items in the smithing table, but nothing happened.

(And the Armor Trim before connecting with the goggle get lost after combining Goggle + Helmet)

**Comment by lukiono** [comment]
FYI: I have now tested it in an instance with only Create: Goggles and Create itself.
When the combined Goggles are crafted using the crafting table, they can no longer be separated in the Smithing Table.

If you use the Smithing Table for the combination, everything can be separated as expected.

**Comment by kkempfer** [comment]
I report the same bug on Minecraft 1.19.2.

Minecraft version: 1.19.2
Mod loader: 0.15.11
Fabric API: 0.76.1
Create Fabric: 0.5.1c
Create Goggles: 0.5.5.g

I confirm @lukiono's latest comment too.

Combined Goggles crafted using the smithing table, then separation in the same smithing table
<img src="https://github.com/user-attachments/assets/e88d55a5-5cb4-45f4-9385-6739053ffab0" height="230">

Combined Goggles crafted using the crafting table, then impossible to separate them in the smithing table
<img src="https://github.com/user-attachments/assets/bff1c9bc-a8aa-4c91-9f0b-7d3ef1c9f3f0" height="230">

**Comment by kkempfer** [comment]
The item IDs and NBT tags are different.
* Their IDs should be the same.
* The helmet created in the smithing table has an extra "CGModifier" tag.

```
/data get entity @p Inventory[{Slot:4b}]
/data get entity @p Inventory[{Slot:6b}]
```

![image](https://github.com/user-attachments/assets/111ecb7e-9a78-4718-ae24-a7fb9ff78a9d)

**Comment by Robocraft999** [comment]
That is correct. As you said their id's are different, the first one is a helmet item from this mod, which was created by crafting and has therefore to be crushed for separation and the second one is a vanilla helmet with a tag, which you remove the same way you applied it. But this gives me an idea, I could add recipes to uncraft the crafted version be letting the result be the goggles and leaving the helmet in the crafting grid like an empty bucket

**Comment by kkempfer** [comment]
I confirm that crushing a goggle helmet that was crafted in a crafting table separates it into a helmet and a goggle. If the goggle helmet was enchanted, all enchantments are lost. On the other hand, crushing a goggle helmet that was build in a smithing table will just break it. In a smithing table, separating a goggle helmet that was build in a smithing table will not remove the enchantments, which is expected.

To be fair, this double crafting / separation mechanic is confusing for the player.

I propose the following  solution:

#### Creation of goggle helmet

I would just remove the crafting recipe using the crafting table as a whole. I would also use the anvil instead of a smithing table. Therefore, the user would only be allowed to craft a goggle helmet in an anvil. I prefer this approach for several reasons:
* From a gameplay perspective, it feels natural to hammer the goggles onto the helmet in an anvil instead of a crafting table.
* Adding goggles onto a helmet is similar to adding an enchantment, i.e. adding extra functionality, which is done in an anvil, not a smithing table.

<img src="https://github.com/user-attachments/assets/6fb1fcec-d78e-4d00-89e9-05e1592277ea" height="230">

#### Separation

I would remove the crushing mechanic as a separation mechanic. It makes more sense to break the goggle helmets if crushed. On the other hand, I like your idea of the modifier remover, because removing goggles from a helmet feels like a precise task that requires patience. The modifier remover tool gives this feeling. However, I would remove the smithing table separation recipe. I have a preference for the standard crafting recipe approach. If I understood your previous message well, it's the same idea you proposed:

Input:
<img src="https://github.com/user-attachments/assets/0ffd0748-3286-4d2b-8ef4-3a3c56b4a0f3" height="230">

Output:
<img src="https://github.com/user-attachments/assets/c426d194-c4c6-4e65-b5a9-5cbd346a4079" height="230">

Cheers

**Comment by Robocraft999** [comment]
>  I would also use the anvil instead of a smithing table.

The Problem is, there is nothing like an anvil recipe. There is no non hacky way to to it sadly, that's because I used the smithing table

> If I understood your previous message well, it's the same idea you proposed:

Correct

The thing is still, there is a difference between the goggle helmet created by crafting and the one by smithing, because the smihing one only applies a modifier (you see it as tag) to the vanilla helmet, or either any helmet, the recipe is more like you can combine any helmet even modded ones with goggles, add a tag to the item, which then renderes the goggles over the item sprite in you inventory and enables you to see the goggle information.
But the Helet in itself doesn't get changed, it is the same item, with same tags, enchants, you name it.
The crafting method on the other hand creates a new item which gets added by the mod. Every helmet you want to have a goggle version would have to be added to the mod, because it is a seperate item. If you craft a normal helmet together with goggles the enchants get copied over to the new item. Because the crusher recipe does't allow for copying over the enchants to the normal helmet after seperation the are lost. If you just remove the tag for the smithing variant, it is as said still the same item, keeps enchants tags etc.
The only reason why I still have'nt removed the seperate goggle helmet items is, because it reduces the content of the mod significantly but I may do it at some time.
I'm also experimenting with the usage of armor templates or upgrades to maybe find a better way of upgrading, We'll see
